File: back/src/main/java/com/back/domain/member/dto/MemberWithUsernameAndWidgetLinkDto.java
package com.back.domain.member.dto;

import com.back.domain.member.entity.Member;
import jakarta.validation.constraints.NotNull;

public record MemberWithUsernameAndWidgetLinkDto(
        @NotNull
        Long id,
        @NotNull
        String username,
        @NotNull
        String githubId,
        @NotNull
        String githubLink,
        @NotNull
        String widgetLink
) {
    public MemberWithUsernameAndWidgetLinkDto(Member member) {
        this(member.getId(),
                member.getUsername(),
                member.getGithubId(),
                member.getGithubLink(),
                member.getWidgetLink());
    }
}

File: back/src/main/java/com/back/domain/member/dto/MemberWithUsernameAndWidgetLinkDto.kt
package com.back.domain.member.dto

import com.back.domain.member.entity.Member

class MemberWithUsernameAndWidgetLinkDto(
    id: Long,
    githubId: String?,
    githubLink: String?,
    val username: String?,
    val widgetLink: String?
) : MemberDto(id, githubId, githubLink) {
    constructor(member: Member) : this(
        member.id,
        member.githubId,
        member.githubLink,
        member.username,
        member.widgetLink
    )
}